tests/dogtag/pytest-ansible/pytest/performance_test/test_cert_enrollment.py

```diff
@@ -71,32 +77,51 @@ def __init__(self, sn_uid):
 
         enrollment_results = cert_client.enroll_cert('caUserCert', inputs)
 
+        if enrollment_results.__len__() == 0:
+            raise Exception("Enrollment results is empty")
+
         for enrollment_result in enrollment_results:
             request_data = enrollment_result.request
             cert_data = enrollment_result.cert
-            print('Request ID: ' + request_data.request_id)
+            if request_data.request_status != 'complete':
+                raise Exception("Cert enrollment failed : {}".format(cert_data))
+            else:
+                self.cert_serial_number=cert_data.serial_number
+
 
 def run_test(sn_uid, number_of_tests_per_thread):
     # execute the specified number of tests sequentially
     for i in range(number_of_tests_per_thread):
-        cert_enroll(sn_uid)
+        serial_num=cert_enroll(sn_uid)
+        cert_list.append(serial_num.cert_serial_number)
```

Comment:
@edewata Should we use something like [a ThreadPool executor](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ThreadPoolExecutor) and use a [map](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Executor.map) and join the result?

Reply:
I'm not too familiar with `ThreadPoolExecutor`, but we can certainly use that if that will clean up the code.

Reply:
Thanks, will make the improvements later